### ISSUE TITLE
Signal parameter is missing from generated entry method

### ISSUE THREAD

**Issue by nboldi** [ELTE-Soft/txtUML#468]
The txtUML code is:
```
	public class Controlled extends State {
		@Override
		public void entry() {
			Signalsignal = getTrigger(Signal.class);
		}
	}
```

The generated code:
```
void MoonLander::Controlled_entry() {
	std::shared_ptr < ControlCycleSignal_EC > temp0 = std::shared_ptr
			< ControlCycleSignal_EC
			> (new ControlCycleSignal_EC(
					static_cast<const ControlCycleSignal_EC&>(e_)));
}
```

It is clear that `e_` should be a parameter of the method as it is for guards:

```
bool MoonLander::guard0(EventBaseCRef e_) { ... }
```

I think it is expected for `getTrigger` to work in entry methods, as its documentation does not say otherwise.

**Comment by djnemeth** [comment]
I believe this is addressed in #439.

**Comment by nagyandras95** [comment]
Are we really want to support this opportunity? A state is able to reached via different transitions so it is not clear what was the signal caused the transition. 

Anyway this code cannot be exported to UML (the signal is unrooted element). Haven't you wrote ControlCycleSignal instead of Signal in the original code?
> 	public class Controlled extends State {
> 		@Override
> 		public void entry() {
> 			ControlCycleSignal signal = getTrigger(ControlCycleSignal .class);
> 		}
> 	}
